test/acceptance/coffee/FlushingDocsTests.coffee

```diff
@@ -55,25 +55,26 @@ describe "Flushing a doc to Mongo", ->
 		it "should not flush the doc to the web api", ->
 			MockWebApi.setDocument.called.should.equal false
 
-	describe "when the web api http request takes a long time", ->
+	describe "when the web api http request takes a long time on first request", ->
 		before (done) ->
 			[@project_id, @doc_id] = [DocUpdaterClient.randomId(), DocUpdaterClient.randomId()]
-			@timeout = 10000
 			MockWebApi.insertDoc @project_id, @doc_id, {
 				lines: @lines
 				version: @version
 			}
+			t = 30000
 			sinon.stub MockWebApi, "setDocument", (project_id, doc_id, lines, version, ranges, callback = (error) ->) ->
-				setTimeout callback, 30000
+				setTimeout callback, t
```

Comment:
You'll have to change this to return a 500 response if we don't retry on timeouts.